src/server/scripts/Northrend/IcecrownCitadel/icecrown_citadel.cpp

```diff
@@ -2155,6 +2171,293 @@ class at_icc_start_frostwing_gauntlet : public AreaTriggerScript
         }
 };
 
+class npc_flesh_eating_insect : public CreatureScript
+{
+public:
+    npc_flesh_eating_insect() : CreatureScript("npc_flesh_eating_insect") { }
+
+    struct npc_flesh_eating_insectAI : public ScriptedAI
+    {
+        npc_flesh_eating_insectAI(Creature* creature) : ScriptedAI(creature)
+        {
+            me->SetReactState(REACT_AGGRESSIVE);
```

Comment:
checking retail videos and the behaviour of the event looked fine to me. I don't know if there is anything i can change here without doing some hacks